orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -116,6 +116,30 @@ class ORTTrainerOptions(object):
                         }
                     }
                 },
+                'recompute': {
+                    'type': 'dict',
+                    'required': False,
+                    'default': {},
+                    'schema': {
+                        'attn_dropout': {
+                            'type': 'boolean',
+                            'default': False
+                        },
+                        'gelu': {
+                            'type': 'boolean',
+                            'default': False
+                        },
+                        'transformer_layer': {
+                            'type': 'boolean',
+                            'default': False
+                        }, 
+                        'number_layers': {
+                            'type': 'integer',
+                            'min': 0,
+                            'default': 0
+                        }
```

Comment:
Can these two options merge into num_transformer_layers? So recompute.num_transformer_layers ==0 means no recompute of layers. Besides, can we use both dropout/gelu recompute and layer recompute?